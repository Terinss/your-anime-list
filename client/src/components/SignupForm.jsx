import React, { Component } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import 'bootstrap/dist/css/bootstrap.min.css';
import { UNSAFE_NavigationContext } from 'react-router-dom';

class SignupForm extends Component {
  constructor(props) {
    super(props);
    this.state = { username: '', password: '' };
    this.createUser = this.createUser.bind(this);
  }

  createUser(event) {
    try {
      event.preventDefault();
      const formData = this.state;
      fetch('/api/users/signup', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify({
          username: formData.username,
          password: formData.password,
        }),
      })
        .then((res) => {
          if (!res.ok) {
            throw new Error('Unable to create new user, invalid input');
          }
        })
        .catch((err) => console.log(err));
    } catch (error) {
      console.error('!', error);
    }
  }

  render() {
    return (
      <Form className="container-lg" onSubmit={this.createUser}>
        <Form.Group className="mb-3" controlId="formBasicUsername">
          <Form.Label>Username</Form.Label>
          <Form.Control
            type="text"
            placeholder="Enter username"
            onChange={(e) =>
              this.setState({
                username: e.target.value,
                password: this.state.password,
              })
            }
          />
        </Form.Group>
        <Form.Group className="mb-3" controlId="formBasicPassword">
          <Form.Label>Password</Form.Label>
          <Form.Control
            type="password"
            placeholder="Enter password"
            onChange={(e) =>
              this.setState({
                username: this.state.username,
                password: e.target.value,
              })
            }
          />
        </Form.Group>
        <Button variant="primary" type="submit">
          Register Account
        </Button>
      </Form>
    );
  }
}

export default SignupForm;
