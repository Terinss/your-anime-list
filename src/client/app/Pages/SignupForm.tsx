import React, { useState } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useNavigate } from 'react-router-dom';
import api from '../components/api/api_instance';
import '../styles/signupform.css';

const SignupForm: React.FC = () => {
  const [formData, setFormData] = useState({ username: '', password: '' });
  const navigate = useNavigate();

  const createUser = (event: React.SyntheticEvent) => {
    try {
      event.preventDefault();
      api
        .post('/api/users/signup', {
          body: JSON.stringify(formData),
        })
        .then(({ data }) => {
          if (!data.ok) {
            throw new Error('Unable to create new user, invalid input');
          } else navigate('/');
        })
        .catch((err) => err);
    } catch (error) {
      console.error('!', error);
    }
  };

  return (
    <Form className="container-lg signup-form" onSubmit={createUser}>
      <h1>Sign Up</h1>
      <Form.Group className="mb-3" controlId="formBasicusername">
        <Form.Label>Username</Form.Label>
        <Form.Control
          data-testid="username"
          type="text"
          placeholder="Enter username"
          onChange={(e) =>
            setFormData({
              username: e.target.value,
              password: formData.password,
            })
          }
        />
      </Form.Group>
      <Form.Group className="mb-3" controlId="formBasicPassword">
        <Form.Label>Password</Form.Label>
        <Form.Control
          data-testid="password"
          type="password"
          placeholder="Enter password"
          onChange={(e) =>
            setFormData({
              username: formData.username,
              password: e.target.value,
            })
          }
        />
      </Form.Group>
      <div className="button-container">
        <Button variant="primary" type="submit" data-testid="register">
          Register Account
        </Button>
        <Button
          variant="secondary"
          type="button"
          data-testid="back-to-login"
          onClick={() => navigate('/login')}
        >
          Back to Login
        </Button>
      </div>
    </Form>
  );
};

export default SignupForm;
