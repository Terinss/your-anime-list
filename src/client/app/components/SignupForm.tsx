import React, { useState } from 'react';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';
import 'bootstrap/dist/css/bootstrap.min.css';
import { useNavigate } from 'react-router-dom';
import '../styles/signupform.css';

const SignupForm: React.FC = () => {
  const [formData, setFormData] = useState({ username: '', password: '' });
  const navigate = useNavigate();

  const createUser = (event: React.SyntheticEvent) => {
    try {
      event.preventDefault();
      fetch('/api/users/signup', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(formData),
      })
        .then((res) => {
          if (!res.ok) {
            throw new Error('Unable to create new user, invalid input');
          } else navigate('/');
        })
        .catch((err) => console.log(err));
    } catch (error) {
      console.error('!', error);
    }
  };

  return (
    <Form className="container-lg signup-form" onSubmit={createUser}>
      <h1>Sign Up</h1>
      <Form.Group className="mb-3" controlId="formBasicUsername">
        <Form.Label>Username</Form.Label>
        <Form.Control
          type="text"
          placeholder="Enter username"
          onChange={(e) =>
            setFormData({
              username: e.target.value,
              password: formData.password,
            })
          }
        />
      </Form.Group>
      <Form.Group className="mb-3" controlId="formBasicPassword">
        <Form.Label>Password</Form.Label>
        <Form.Control
          type="password"
          placeholder="Enter password"
          onChange={(e) =>
            setFormData({
              username: formData.username,
              password: e.target.value,
            })
          }
        />
      </Form.Group>
      <Button variant="primary" type="submit">
        Register Account
      </Button>
    </Form>
  );
};

export default SignupForm;
