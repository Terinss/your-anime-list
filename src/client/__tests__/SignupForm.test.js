import { render } from '@testing-library/react';
import SignupForm from '../app/components/SignupForm';
import jest from 'jest';
import { MemoryRouter } from 'react-router-dom';

test('SignupForm renders correctly', () => {
  const { container } = render(
    <MemoryRouter>
      <SignupForm />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
