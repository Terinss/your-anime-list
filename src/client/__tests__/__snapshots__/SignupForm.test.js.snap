// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignupForm renders correctly 1`] = `
<div>
  <form
    class="container-lg signup-form"
  >
    <h1>
      Sign Up
    </h1>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="formBasicUsername"
      >
        Username
      </label>
      <input
        class="form-control"
        id="formBasicUsername"
        placeholder="Enter username"
        type="text"
      />
    </div>
    <div
      class="mb-3"
    >
      <label
        class="form-label"
        for="formBasicPassword"
      >
        Password
      </label>
      <input
        class="form-control"
        id="formBasicPassword"
        placeholder="Enter password"
        type="password"
      />
    </div>
    <button
      class="btn btn-primary"
      type="submit"
    >
      Register Account
    </button>
  </form>
</div>
`;
